src/github.com/couchbase/sync_gateway/base/bucket_gocb.go

```diff
@@ -607,13 +611,31 @@ func (bucket CouchbaseBucketGoCB) GetAndTouchRaw(k string, exp int) (rv []byte,
 }
 
 func (bucket CouchbaseBucketGoCB) Add(k string, exp int, v interface{}) (added bool, err error) {
-	LogPanic("Unimplemented method: Add()")
-	return false, nil
+	bucket.singleOps <- struct{}{}
+	defer func() {
+		<-bucket.singleOps
+	}()
+	gocbExpvars.Add("Add", 1)
+	_, err = bucket.Bucket.Insert(k, v, uint32(exp))
+
+	if GoCBErrorType(err) == GoCBErr_MemdStatusKeyExists {
+		return false, nil
+	}
+	return err == nil, err
 }
 
 func (bucket CouchbaseBucketGoCB) AddRaw(k string, exp int, v []byte) (added bool, err error) {
-	LogPanic("Unimplemented method: AddRaw()")
-	return false, nil
+	bucket.singleOps <- struct{}{}
+	defer func() {
+		<-bucket.singleOps
+	}()
```

Comment:
It's used to manage the total number of gocb ops we run concurrently, to avoid gocb queue overflow.

I agree this could be refactored in a way that makes it's function clearer throughout bucket_gocb, but would prefer to wait for 1.3 for that refactoring.